datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadbalancingStrategy.java

```diff
@@ -101,14 +101,19 @@ private List<DatastreamTask> getDatastreamTasks(List<DatastreamGroup> datastream
     List<DatastreamTask> currentlyAssignedDatastreamTasks = new ArrayList<>();
     currentAssignment.values().forEach(currentlyAssignedDatastreamTasks::addAll);
 
+    Set<String> uniqueDatastreamTaskNames = new HashSet<>();
     for (DatastreamGroup dg : datastreams) {
       Set<DatastreamTask> tasksForDatastreamGroup = currentlyAssignedDatastreamTasks.stream()
-          .filter(x -> x.getTaskPrefix().equals(dg.getTaskPrefix()))
+          .filter(x -> x.getTaskPrefix().equals(dg.getTaskPrefix()) && uniqueDatastreamTaskNames.add(x.getDatastreamTaskName()))
           .collect(Collectors.toSet());
```

Comment:
Since we are collecting all of them in a set anyway, why do we need a separate set to handle this?

Reply:
Checking task name equality is a more general check than checking equality of two datastreams which will check a suite of fields associated with the datastreams

Reply:
I am good with either of those. But my only concern is that the test you have won't cover the case when the datastream tasks have different internal fields and the same task names. And that test would pass irrespective of the change you have in this PR for load balancing strategy.

But in my opinion, we would never hit that scenario for this issue since the previous leader would have copied the task to the other instance entirely, so all the internal fields should also match anyways.

Reply:
Added tests for different internal fields